packages/components/src/ListView/Items/Item/Item.component.js

```diff
@@ -8,58 +8,71 @@ function itemLabelClasses() {
-			const higlighted = part.toLowerCase() === searchCriteria.toLowerCase() ? <b>{part}</b> : part;
-			return <span key={index}>{higlighted}</span>;
-		});
+class Item extends Component {
+	componentWillReceiveProps(nextProps) {
+		const { children, measure } = this.props;
+		const { children: nextChildren } = nextProps;
+		if (children && children.length !== nextChildren.length) {
+			measure();
+		}
 	}
 
-	const itemId = id && `checkbox-${id}`;
-	const itemClassName = classNames(
-		'checkbox',
-		{ 'switch-nested': props.children },
-		{ switch: isSwitchBox },
-	);
+	render() {
+		const { id, item, parentItem, isSwitchBox, searchCriteria, children, measure } = this.props;
+
+		/**
+		 * This function allow to get component rendering based on searchCriteria
+		 * @param label the current label to parse & to render
+		 */
+		function getSearchedLabel(label) {
+			const splitRegex = new RegExp(`(${searchCriteria})`, 'gi');
+			return label.split(splitRegex).map((part, index) => {
+				const higlighted =
+					part.toLowerCase() === searchCriteria.toLowerCase() ? <b>{part}</b> : part;
+				return <span key={index}>{higlighted}</span>;
+			});
+		}
 
-	return (
-		<div id={id}>
-			<div className="checkbox-container">
-				<div className={itemClassName} key={item.index}>
-					<label htmlFor={itemId}>
-						<input
-							id={itemId}
-							type="checkbox"
-							checked={item.checked}
-							onChange={event => item.onChange(event, item, parentItem)}
-						/>
-						<span className={itemLabelClasses()}>
-							{searchCriteria ? getSearchedLabel(item.label) : item.label}
-						</span>
-					</label>
-					{props.children && (
-						<div className={classNames('checkbox-nested-expand', { expanded: item.expanded })}>
-							<Action
-								bsStyle="link"
-								icon="talend-caret-down"
-								onClick={event => item.onExpandToggle(event, item)}
+		const itemId = id && `checkbox-${id}`;
+		const itemClassName = classNames(
+			'checkbox',
+			{ 'switch-nested': children },
+			{ switch: isSwitchBox },
+		);
+
+		return (
+			<div id={id}>
+				<div className="checkbox-container">
+					<div className={itemClassName} key={item.index}>
+						<label htmlFor={itemId}>
+							<input
+								id={itemId}
+								type="checkbox"
+								checked={item.checked}
+								onChange={event => item.onChange(event, item, parentItem)}
 							/>
-						</div>
-					)}
-					{props.children &&
-						item.expanded && <div className="checkbox-nested">{props.children}</div>}
+							<span className={itemLabelClasses()}>
+								{searchCriteria ? getSearchedLabel(item.label) : item.label}
+							</span>
+						</label>
+						{children && (
+							<div className={classNames('checkbox-nested-expand', { expanded: item.expanded })}>
+								<Action
+									bsStyle="link"
+									icon="talend-caret-down"
+									onClick={event => {
+										item.onExpandToggle(event, item);
+										measure();
```

Comment:
I don't think this will work as the Action will probably have some async steps (React render).
Moreover, you added a `componenWillReceiveProps`. I think you can remove this `measure()` on action trigger.